packages/tools/src/tools/annotation/EllipticalROITool.ts

```diff
@@ -1116,57 +1092,29 @@ class EllipticalROITool extends AnnotationTool {
           scale /
           scale;
 
-        let count = 0;
-        let mean = 0;
-        let stdDev = 0;
-        let max = -Infinity;
-
-        const meanMaxCalculator = ({ value: newValue }) => {
-          if (newValue > max) {
-            max = newValue;
-          }
-
-          mean += newValue;
-          count += 1;
-        };
-
-        pointInShapeCallback(
-          imageData,
-          (pointLPS, pointIJK) => pointInEllipse(ellipseObj, pointLPS),
-          meanMaxCalculator,
-          boundsIJK
+        const modalityUnit = getModalityUnit(
+          metadata.Modality,
+          annotation.metadata.referencedImageId,
+          modalityUnitOptions
         );
 
-        mean /= count;
-
-        const stdCalculator = ({ value }) => {
-          const valueMinusMean = value - mean;
-
-          stdDev += valueMinusMean * valueMinusMean;
-        };
-
-        pointInShapeCallback(
+        const pointsInShape = pointInShapeCallback(
           imageData,
           (pointLPS, pointIJK) => pointInEllipse(ellipseObj, pointLPS),
-          stdCalculator,
+          null,
           boundsIJK
         );
 
-        stdDev /= count;
-        stdDev = Math.sqrt(stdDev);
-
-        const modalityUnit = getModalityUnit(
-          metadata.Modality,
-          annotation.metadata.referencedImageId,
-          modalityUnitOptions
-        );
+        const stats = this.configuration.statsCalculator(pointsInShape);
```

Comment:
so seems like there is some optimization issue happening here. 
Previously we had `pointInShapeCallback` which would take care of looping over the points in the shape and running a callback (e.g., meanMaxCalculator), but now we run the point in shape to grab the points and then again in the statsCalculator to run over them again (two loops instead of one), can you just make the statsCalculator a callbackto the pointInShape so that we don't run twice?

Reply:
So if I want to use a callback to calculate the stats directly into the `pointInShapeCallback `, I would need to make my statsCalculator a function of a class that stores the sum of points and could return the calculated stats (such as `meanMaxCalculator`).

Does that seems correct to you ?

Reply:
I think so